internal/nodes/avi_model_l4_translator.go

```diff
@@ -47,6 +47,7 @@ func (o *AviObjectGraph) ConstructAviL4VsNode(svcObj *corev1.Service, key string
 	}
 
 	if extDNS, ok := svcObj.Annotations[lib.ExternalDNSAnnotation]; ok && autoFQDN {
+		autoFQDN = false
```

Comment:
@akshayhavile FYI, I will be moving the change related to externalDNS annotation with autoFQDN to a separate PR.